Packs/Base/Scripts/DBotPredictURLPhishing/DBotPredictURLPhishing.py

```diff
@@ -0,0 +1,588 @@
+        + WEIGHT_HEURISTIC[MODEL_KEY_SEO] + WEIGHT_HEURISTIC[MODEL_KEY_URL_SCORE] \
+        + WEIGHT_HEURISTIC[MODEL_KEY_LOGO_FOUND] * use_logo
+    score = (use_age * WEIGHT_HEURISTIC[DOMAIN_AGE_KEY] * domain_age_key
+             + WEIGHT_HEURISTIC[MODEL_KEY_LOGIN_FORM] * pred_json[MODEL_KEY_LOGIN_FORM]
+             + WEIGHT_HEURISTIC[MODEL_KEY_SEO] * pred_json[MODEL_KEY_SEO]
+             + WEIGHT_HEURISTIC[MODEL_KEY_URL_SCORE] * pred_json[MODEL_KEY_URL_SCORE]
+             + use_logo * WEIGHT_HEURISTIC[MODEL_KEY_LOGO_FOUND] * pred_json[MODEL_KEY_LOGO_FOUND]) / total_weight_used
+    return score
+
+
+def get_verdict(pred_json: Dict, is_white_listed: bool) -> Tuple[float, str]:
+    """
+    Return verdict of the url based on the output of the model
+    :param pred_json: output from the model
+    :return:
+    """
+    if is_white_listed:
+        return SCORE_BENIGN, BENIGN_VERDICT
+    score = get_score(pred_json)
+    if pred_json[MODEL_KEY_LOGO_FOUND]:
+        return score, MALICIOUS_VERDICT
+    else:
+        if score < BENIGN_THRESHOLD:
+            return score, BENIGN_VERDICT
+        elif score < SUSPICIOUS_THRESHOLD:
+            return score, SUSPICIOUS_VERDICT
+        else:
+            return score, MALICIOUS_VERDICT
+
+
+def create_dict_context(url, verdict, pred_json, score, is_white_listed, output_rasterize):
+    return {'url': url, 'verdict': verdict, 'pred_json': pred_json, 'score': score, 'is_white_listed': is_white_listed,
+            'output_rasterize': output_rasterize}
+
+
+def extract_created_date(entry_list: List):
+    """
+    Check if domain age is younger than THRESHOLD_NEW_DOMAIN_YEAR year
+    :param entry_list: output of the whois command
+    :return: bool
+    """
+    for entry in entry_list:
+        if is_error(entry):
+            continue
+        else:
+            date_str = entry['EntryContext'].get('Domain(val.Name && val.Name == obj.Name)', {}).get('WHOIS', {}).get(
+                'CreationDate', None)
+            if date_str:
+                date = datetime.strptime(date_str, '%d-%m-%Y')
+                threshold_date = datetime.now() - timedelta(days=THRESHOLD_NEW_DOMAIN_MONTHS * 30)
+                return date > threshold_date
+    return None
+
+
+def get_prediction_single_url(model, url, force_model):
+    is_white_listed = False
+    valid_url, error = is_valid_url(url)
+
+    if not valid_url:
+        return create_dict_context(url, MSG_INVALID_URL + error, {}, SCORE_INVALID_URL, is_white_listed, {})
+
+    # Check domain age from WHOIS command
+    domain = extract_domainv2(url)
+    res = demisto.executeCommand('whois', {'query': domain,
+                                           })
+    is_new_domain = extract_created_date(res)
+
+    # Check is domain in white list -  If yes we don't run the model
+    if in_white_list(model, url):
+        if not force_model:
+            return create_dict_context(url, BENIGN_VERDICT_WHITELIST, {}, SCORE_BENIGN, is_white_listed, {})
```

Comment:
This flow is a bit problematic, as a lot of keys are missing from context compared to when the model is forced. Please add all keys from full entry, and you can specify a `Unknown` value. also need a field for verdict which is not human readable so it'd be easy to work with, just the verdict (without formatting)
![image](https://user-images.githubusercontent.com/57318201/135424636-efa0f925-a78e-4cba-9a2f-876307bb7bd9.png)

Reply:
This is the first entry we return with only 2 fields. There are others entries with details which should contains same fields. This is the code:

def return_entry_white_list(url):
    """
    Create syntethci entry when url belongs to whitelist
    :param url: url
    :return:
    """
    explain = {
        KEY_CONTENT_DOMAIN: extract_domainv2(url),
        KEY_CONTENT_URL: url,
        KEY_CONTENT_AGE: MSG_WHITE_LIST,
        KEY_CONTENT_LOGO: MSG_WHITE_LIST,
        KEY_CONTENT_LOGIN: MSG_WHITE_LIST,
        KEY_CONTENT_URL_SCORE: MSG_WHITE_LIST,
        KEY_CONTENT_SEO: MSG_WHITE_LIST
    }
    explain_hr = {
        "Domain": extract_domainv2(url),
        "Has the domain good SEO ?": MSG_WHITE_LIST,
        DOMAIN_AGE_KEY: MSG_WHITE_LIST,
        "Has the website a login page?": MSG_WHITE_LIST,
        "Logo found that does not correspond to the given domain": MSG_WHITE_LIST,
        "URL severity score (from 0 to 1)": MSG_WHITE_LIST
    }
    verdict_hr = {
        "Verdict": BENIGN_VERDICT,
        "URL": url
    }
    return_entry = {
        "Type": entryTypes["note"],
        "ContentsFormat": formats['json'],
        "HumanReadable": tableToMarkdown("Verdict", verdict_hr) + tableToMarkdown("Report", explain_hr),
        "Contents": explain,
        "EntryContext": {'DBotPredictURLPhishing': explain}
    }
    demisto.results(return_entry)